### PR TITLE
osmium-tool: init at 1.9.1 incl. dependencies

### PR DESCRIPTION
**Pull request by das-g**
###### Motivation for this change

I'd like to use the osmium CLI tool for analyzing OpenStreetMap data or for preprocessing it for use in other analyzing software.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix run nixpkgs.nox -c nox-review wip --against origin/master`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Note that the various [optional dependencies of libosmium](https://osmcode.org/libosmium/manual.html#extra-dependencies) aren't supported by this PR, yet.

### CODE REVIEW

**Comment by srhb** [comment]
@grahamcofborg build protozero libosmium osmium-tool

File: pkgs/development/libraries/libosmium/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ protozero zlib bzip2 expat boost ];
+
+
+  meta = {
+    description = "Fast and flexible C++ library for working with OpenStreetMap data";
+    homepage = "https://osmcode.org/libosmium/";
+    licence = stdenv.lib.licences.boost;
+    maintainers = with stdenv.maintainers; [ das-g ];
```
Comment (srhb) [inline pkgs/development/libraries/libosmium/default.nix]: `stdenv.lib.maintainers`

File: pkgs/applications/misc/osmium-tool/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ libosmium protozero boost bzip2 zlib expat ];
+
+  meta = {
+    description = "Multipurpose command line tool for working with OpenStreetMap data based on the Osmium library";
+    homepage = "https://osmcode.org/osmium-tool/";
+    license = with stdenv.lib.licenses; [ gpl3 mit bsd3 ];
+    maintainers = with stdenv.maintainers; [ das-g ];
```
Comment (srhb) [inline pkgs/applications/misc/osmium-tool/default.nix]: `stdenv.lib.maintainers`

**Review by srhb — CHANGES_REQUESTED** [review]

**Comment by das-g** [comment]
How can I reproduce the failures repored by @GrahamcOfBorg locally?

Neither
```bash
nix build -f $PATH_TO_LOCAL_nixpkgs_CLONE/ osmium-tool
```
nor
```bash
cd $PATH_TO_LOCAL_nixpkgs_CLONE/
nix run nixpkgs.nox -c nox-review wip --against origin/master
```
gave me these errors.

File: pkgs/development/libraries/libosmium/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ protozero zlib bzip2 expat boost ];
+
+
+  meta = with stdenv.lib; {
+    description = "Fast and flexible C++ library for working with OpenStreetMap data";
+    homepage = "https://osmcode.org/libosmium/";
+    licence = licences.boost;
```
Comment (das-g) [inline pkgs/development/libraries/libosmium/default.nix]: typo: should be "licen**s**es"

Comment (das-g) [reply pkgs/development/libraries/libosmium/default.nix]: Should I fix this in a new commit, or &mdash; as this was never a valid meta attributes set &mdash; rewrite the commit that introduced the typo?

Comment (srhb) [reply pkgs/development/libraries/libosmium/default.nix]: Feel free to squash it into the existing commit. I'd prefer we end up with just the four, one for each application and the maintainer addition. :)

**Comment by srhb** [comment]
@GrahamcOfBorg build protozero libosmium osmium-tool

**Review by srhb — APPROVED** [review]

**Pull request merged**